### PR TITLE
Altukhov/search and follow feature

### PR DESCRIPTION
**Pull request by Legendorik**
изображение
изображение
изображение

### CODE REVIEW

**Comment by Legendorik** [comment]
https://github.com/AlexRyzhickov/Epam-Internship-project/pull/2/commits/28917859729f5264a92da04edb76ec90fc348572
Исправлены замечания @AlexRyzhickov

**Comment by AlexRyzhickov** [comment]
image
 I can follow myself

File: app/src/main/java/com/epam/bet/viewmodel/SubscribeViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        } else {
+            context?.showToast("Problem with connection !")
+        }
+        return false
+    }
+
+    fun addIFollow(email: String, successListener: (email: String) -> Unit, failListener: () -> Unit) {
+        val myEmail = sharedPreferences.getString("email", "none")
+        val myName = sharedPreferences.getString("name", "none")
+
```
Comment (AlexRyzhickov) [inline app/src/main/java/com/epam/bet/viewmodel/SubscribeViewModel.kt:51]: myEmail, myName -> bullshit naming

File: app/src/main/java/com/epam/bet/viewmodel/SubscribeViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+class SubscribeViewModel(application: Application) : AndroidViewModel(application) {
+
+    private val db = FirebaseFirestore.getInstance()
+    private val users = db.collection("users")
+    private lateinit var sharedPreferences: SharedPreferences
+
+    fun isItMyEmail(email: String): Boolean {
+        sharedPreferences = getApplication<Application>().applicationContext.getSharedPreferences("BetAppSettings", Context.MODE_PRIVATE)
+        val myEmail = sharedPreferences.getString("email", "none")
```
Comment (AlexRyzhickov) [inline app/src/main/java/com/epam/bet/viewmodel/SubscribeViewModel.kt]: Duplicate code (repeated requests to sharedPreferences) 
Put this in a separate function, please !

File: app/src/main/java/com/epam/bet/viewmodel/SubscribeViewModel.kt (hunk 1 above)
Comment (Legendorik) [reply app/src/main/java/com/epam/bet/viewmodel/SubscribeViewModel.kt:51]: Why? We have concepts like "I follow" in the code. In this context, it is clear that "my email" equals "current user email"

**Comment by Legendorik** [comment]
By the way, this is a Russian repository with Russian-speaking participants. Why are you talking to us in English...

File: app/src/main/java/com/epam/bet/fragments/BetsFragment.kt [hunk 3]
```diff
@@ -1,24 +1,82 @@
...
+import com.epam.bet.interfaces.RecyclerViewClickListener
+import com.epam.bet.viewmodel.BetsViewModel
 import com.google.android.material.floatingactionbutton.FloatingActionButton
+import org.koin.android.ext.android.get
 
 //https://epam.sharepoint.com/sites/EPAMSummerStudents2021/Shared%20Documents/General/Mocks/New%20Wireframe%202.png
 
-class BetsFragment : Fragment(R.layout.bets_fragment_layout) {
+class BetsFragment : Fragment(R.layout.bets_fragment_layout), RecyclerViewClickListener {
+    private var _binding: BetsFragmentLayoutBinding? = null
```
Comment (liliyamarkova) [inline app/src/main/java/com/epam/bet/fragments/BetsFragment.kt]: please don't use name with _

File: app/src/main/java/com/epam/bet/fragments/BetsFragment.kt [hunk 4]
```diff
@@ -1,24 +1,82 @@
...
+
+        _binding = BetsFragmentLayoutBinding.inflate(inflater, container, false)
+        betsList = binding.BetsList
+
+
+        //val viewModel = ViewModelProvider(this).get(BetsViewModel::class.java)
+        val viewModel =  get<BetsViewModel>()
+        viewModel.fetchUser()
+        //viewModel.addBet()
+
```
Comment (liliyamarkova) [inline app/src/main/java/com/epam/bet/fragments/BetsFragment.kt:43]: why we store commented code?

File: app/src/main/java/com/epam/bet/fragments/FollowersFragment.kt [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    private var _binding: FollowersFragmentBinding? = null
+    private val binding get() = _binding!!
+    private lateinit var followersAdapter: UserDataAdapter
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        _binding = FollowersFragmentBinding.inflate(inflater, container, false)
```
Comment (liliyamarkova) [inline app/src/main/java/com/epam/bet/fragments/FollowersFragment.kt]: naming

File: app/src/main/java/com/epam/bet/fragments/ProfileFragment.kt [hunk 6]
```diff
@@ -19,6 +23,15 @@ class ProfileFragment : Fragment(R.layout.profile_fragment) {
     private var _binding: ProfileFragmentBinding? = null
     private val binding get() = _binding!!
     private lateinit var authInterface: AuthInterface
+    private lateinit var adapter: ProfileTabsAdapter
+    private lateinit var viewPager: ViewPager2
+    private lateinit var tabLayout: TabLayout
+
+    private val tabNames: Array<String> = arrayOf(
+        "Followers",
```
Comment (liliyamarkova) [inline app/src/main/java/com/epam/bet/fragments/ProfileFragment.kt]: let's use such strings like getResources().getString(R.string.blablabla)

File: app/src/main/java/com/epam/bet/viewmodel/BetsViewModel.kt [hunk 7]
```diff
@@ -1,14 +1,69 @@
...
+    var activeUser: MutableLiveData<User> = MutableLiveData()
+    var betsList: MutableLiveData<List<Bet?>> = MutableLiveData()
+    private lateinit var sharedPreferences : SharedPreferences
+
+
+    fun fetchUser(){
+        sharedPreferences = getApplication<Application>().applicationContext.getSharedPreferences("BetAppSettings", Context.MODE_PRIVATE)
+        val user: User = User()
+        var newBetList: MutableList<Bet> = mutableListOf()
+        val email = sharedPreferences.getString("email", "none")
```
Comment (liliyamarkova) [inline app/src/main/java/com/epam/bet/viewmodel/BetsViewModel.kt:30]: please don't store hardcoded strings

File: app/src/main/java/com/epam/bet/viewmodel/BetsViewModel.kt [hunk 8]
```diff
@@ -1,14 +1,69 @@
...
+        if(email!="none"){
+            users.document(email!!).get().addOnSuccessListener { document ->
```
Comment (liliyamarkova) [inline app/src/main/java/com/epam/bet/viewmodel/BetsViewModel.kt:32]: that's bad idea to use !! instead of ?. and additional checking, let's change?

**Review by liliyamarkova — APPROVED** [review]

**Pull request merged**